zkevm-circuits/src/tx_circuit.rs

```diff
@@ -695,6 +754,44 @@ impl<F: Field> SubCircuitConfig<F> for TxCircuitConfig<F> {
             cb.gate(meta.query_fixed(q_enable, Rotation::cur()))
         });
 
+        // prev block's cum_num_txs < tx_id
+        let tx_id_gt_prev_cnt = LtChip::configure(
+            meta,
+            |meta| meta.query_fixed(q_enable, Rotation::cur()),
+            |meta| {
+                let num_txs = meta.query_advice(num_txs, Rotation::cur());
+                let cum_num_txs = meta.query_advice(cum_num_txs, Rotation::cur());
+
+                cum_num_txs - num_txs
+            },
+            |meta| meta.query_advice(tx_table.tx_id, Rotation::cur()),
+            u8_table.into(),
+        );
+
+        // last non-padding tx must have tx_id == cum_num_txs
+        meta.create_gate(
```

Comment:
will these 2 gates still ok since we need to handle the l1 tx queue index design now?

Reply:
yes, the l1 queue index is only relevant to `num_all_txs`.